### PR TITLE
Change Proxmox `agent` argument to string.

### PR DESCRIPTION
**Pull request by reitermarkus**
##### SUMMARY

Allow specifying Proxmox `agent` argument as string.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`proxmox_kvm`

##### ADDITIONAL INFORMATION

The `agent` argument can not only be `true` or `false` (or `0`/`1`) but also be a string, e.g. `enabled=1,fstrim_cloned_disks=1`.

Extracted from https://github.com/ansible-collections/community.general/pull/4027.

### CODE REVIEW

File: plugins/modules/cloud/misc/proxmox_kvm.py
Comment (reitermarkus) [inline plugins/modules/cloud/misc/proxmox_kvm.py:967]: According to https://github.com/ansible-collections/community.general/pull/4027#discussion_r913231029 these are never true since the type is not a string. Is there a way to make the type support either a boolean or a string.

File: plugins/modules/cloud/misc/proxmox_kvm.py
Comment (reitermarkus) [inline plugins/modules/cloud/misc/proxmox_kvm.py:967]: According to https://github.com/ansible-collections/community.general/pull/4027#discussion_r913231029 these are never true since the type is now a string. Is there a way to make the type support either a boolean or a string.

Comment (reitermarkus) [reply plugins/modules/cloud/misc/proxmox_kvm.py:967]: Or do I just change these to a string comparison with `'True'` and `'False'`?

File: plugins/modules/cloud/misc/proxmox_kvm.py
Comment (felixfontein) [inline plugins/modules/cloud/misc/proxmox_kvm.py:1049]:
```suggestion
        agent=dict(type='str'),
```

File: plugins/modules/cloud/misc/proxmox_kvm.py
Comment (felixfontein) [inline plugins/modules/cloud/misc/proxmox_kvm.py:964]: `agent` is a string argument, so anything passed in will be converted to a string. So now you have to check for *anything* that Ansible accepts as a truish boolean.

You can use something like
```
from ansible.module_utils.parsing.convert_bool import boolean

...

try:
    if boolean(kwargs['agent'], strict=True):
        kwargs['agent'] = 'enabled=1'
    else:
        kwargs['agent'] = 'enabled=0'
except TypeError:
    # Not something that Ansible would parse as a boolean
    pass
```
to handle this correctly.

File: plugins/modules/cloud/misc/proxmox_kvm.py
Comment (felixfontein) [inline plugins/modules/cloud/misc/proxmox_kvm.py:27]:
```suggestion
      - Specify if the QEMU Guest Agent should be enabled/disabled.
      - Since community.general 5.5.0, this can also be a string instead of a boolean.
        This allows to specify values such as C(enabled=1,fstrim_cloned_disks=1).
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: plugins/modules/cloud/misc/proxmox_kvm.py
Comment (reitermarkus) [reply plugins/modules/cloud/misc/proxmox_kvm.py]: Thanks, I added this now.

File: plugins/modules/cloud/misc/proxmox_kvm.py
Comment (russoz) [reply plugins/modules/cloud/misc/proxmox_kvm.py]: The comment referred to the fact that that PR was changing the parameter type, same as this PR is doing. Now, IMO, it makes no sense to change it to `str` and then expect `True` or `False`.

Comment (felixfontein) [reply plugins/modules/cloud/misc/proxmox_kvm.py]: If you passed in `"yes"`, `"1"`, `"on"`, `"true"`, or several other strings to the old version, they were also interpreted as `True`. So to be backwards compatible just checking for `"True"` is not suficient.

File: plugins/modules/cloud/misc/proxmox_kvm.py
Comment (felixfontein) [inline plugins/modules/cloud/misc/proxmox_kvm.py:971]: Is a JSON string `enabled=0` equivalent to a JSON `false` value for all supported versions of proxmoxer / the API? (The same for `enabled=1` and `true`.)

Comment (reitermarkus) [reply plugins/modules/cloud/misc/proxmox_kvm.py:971]: I'm not quite sure. May be best to pass the boolean directly as before.

**Review by felixfontein — APPROVED** [review]
From my understanding of the code, it looks ok. Would be glad if someone who actually uses the module would look at this as well though, or even try it out :)

**Review by russoz — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
If nobody from the maintainers complains, I'll merge somewhen this weekend.

**Comment by felixfontein** [comment]
@reitermarkus thanks for improving this!
@russoz thanks for reviewing!

**Comment by felixfontein** [comment]
@fritterhoff that's rather old news, since that bug was fixed pretty shortly after this got merged (in #5198). Or do you still have this (exact) problem with the latest release?

**Comment by fritterhoff** [comment]
Nvm. Error seems to be fixed. Sorry :(

**Pull request merged**
